File: src/50.py
K = 1_000_000
p = [True]*(K+1)
p[0]=p[1]=False
for i in range(K):
    if p[i]:
        for j in range(i*2,K,i):
            p[j] = False

primes_s = {i for i,v in enumerate(p) if v}
primes_l = [_ for _,__ in enumerate(p) if __]

answer = -1
min_ = 0
len_=len(primes_s)

for i in range(len_):
    for j in range(i+min_,len_):
        k=sum(primes_l[i:j])
        if k>=K:
            break
        if k in primes_s:
            print(min_,i,j,k)
            min_ = j-i
            answer = k
print(answer)

File: src/50-59/50.c
#include <stdio.h>
#include <stdbool.h>

#define K 1000000

int main() {
    bool p[K + 1];
    int primes_l[K + 1];
    int answer = -1;
    int min_ = 0;
    int len_;

    // Sieve of Eratosthenes to generate primes
    for (int i = 0; i <= K; i++)
        p[i] = true;
    p[0] = p[1] = false;
    for (int i = 0; i < K; i++) {
        if (p[i]) {
            for (int j = i * 2; j < K; j += i)
                p[j] = false;
        }
    }

    // Store prime numbers in primes_l
    int idx = 0;
    for (int i = 0; i <= K; i++) {
        if (p[i]) {
            primes_l[idx++] = i;
        }
    }
    len_ = idx;

    // Finding the longest sum of consecutive primes below K
    for (int i = 0; i < len_; i++) {
        for (int j = i + min_; j < len_; j++) {
            int sum = 0;
            for (int k = i; k < j; k++) {
                sum += primes_l[k];
                if (sum >= K)
                    break;
            }
            if (sum >= K)
                break;
            if (p[sum]) {
                min_ = j - i;
                answer = sum;
            }
        }
    }
    printf("%d\n", answer);
    return 0;
}
